rhel/vex/test_common.go

```diff
@@ -0,0 +1,74 @@
+package vex
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/textproto"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"golang.org/x/tools/txtar"
+)
+
+func parseFilenameHeaders(data []byte) (string, http.Header, error) {
+	pf, h, _ := bytes.Cut(data, []byte{' '})
+	compressedFilepath := bytes.TrimSuffix(pf, []byte{'\n'})
+	h = bytes.ReplaceAll(h, []byte(`\n`), []byte{'\n'})
+	// Do headers
+	tp := textproto.NewReader(bufio.NewReader(bytes.NewReader(h)))
+	hdr, err := tp.ReadMIMEHeader()
+	if err != nil && err != io.EOF {
+		return "", nil, err
+	}
+	return string(compressedFilepath), http.Header(hdr), nil
+}
+
+func ServeSecDB(t *testing.T, txtarFile string) (string, *http.Client) {
+	mux := http.NewServeMux()
+	archive, err := txtar.ParseFile(txtarFile)
+	if err != nil {
+		t.Fatal(err)
+	}
+	relFilepath, headers, err := parseFilenameHeaders(archive.Comment)
+	if err != nil {
+		t.Fatal(err)
+	}
+	filename := filepath.Base(relFilepath)
+	mux.HandleFunc("/"+filename, func(w http.ResponseWriter, _ *http.Request) {
+		for k, v := range headers {
+			w.Header().Set(k, v[0])
+		}
+
```

Comment:
guessing `MethodHead` is never used since it was removed from here?

Reply:
The `ServeMux` routes the HEAD method by default (as long as the pattern doesn't exclude it) so having the switch isn't really necessary.

Reply:
Turns out it was necessary, I was trying to track down why tests were flaking with:
```
HTTP/1.x transport connection broken: malformed HTTP response "..."
```
Then when debugging that I found a warning from a previous call:
```
Unsolicited response received on idle HTTP channel starting with "..."
```
So yeah, we shouldn't be sending anything back for HEAD requests, the `ServeMux` won't automatically do that.